app/scripts/modules/core/src/function/Functions.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import { Debounce } from 'lodash-decorators';
+import { Subscription } from 'rxjs';
+import { Application } from 'core/application/application.model';
+import { FilterTags, IFilterTag } from 'core/filterModel/FilterTags';
+import { IFunctionGroup } from 'core/domain';
+import { FunctionState } from 'core/state';
+import { FunctionPod } from './FunctionPod';
+import { Spinner } from 'core/widgets/spinners/Spinner';
+
+export interface IFunctionsProps {
+  app: Application;
+}
+
+export interface IFunctionsState {
+  initialized: boolean;
+  groups: IFunctionGroup[];
+  tags: IFilterTag[];
+}
+
+export class Functions extends React.Component<IFunctionsProps, IFunctionsState> {
+  private groupsUpdatedListener: Subscription;
+  private functionsRefreshUnsubscribe: () => any;
+
+  constructor(props: IFunctionsProps) {
+    super(props);
+    this.state = {
+      initialized: false,
+      groups: [],
+      tags: [],
+    };
+  }
+
+  public componentDidMount(): void {
+    const { app } = this.props;
+
+    this.groupsUpdatedListener = FunctionState.filterService.groupsUpdatedStream.subscribe(() => this.groupsUpdated());
+    FunctionState.filterModel.asFilterModel.activate();
+    this.functionsRefreshUnsubscribe = app
+      .getDataSource('functions')
+      .onRefresh(null, () => this.updateFunctionGroups());
+    app.setActiveState(app.loadBalancers);
+    this.updateFunctionGroups();
+  }
+
+  public componentWillUnmount(): void {
+    this.groupsUpdatedListener.unsubscribe();
+    this.functionsRefreshUnsubscribe();
+  }
+
+  private groupsUpdated(): void {
+    this.setState({
+      groups: FunctionState.filterModel.asFilterModel.groups,
+      tags: FunctionState.filterModel.asFilterModel.tags,
+    });
+  }
+
+  @Debounce(200)
+  private updateFunctionGroups(): void {
+    FunctionState.filterModel.asFilterModel.applyParamsToUrl();
+    FunctionState.filterService.updateFunctionGroups(this.props.app);
+    this.groupsUpdated();
+
+    if (this.props.app.getDataSource('functions').loaded) {
+      this.setState({ initialized: true });
+    }
+  }
+
+  private clearFilters = (): void => {
+    FunctionState.filterService.clearFilters();
+    this.updateFunctionGroups();
+  };
+
+  private tagCleared = (): void => {
+    this.updateFunctionGroups();
+  };
+
+  public render(): React.ReactElement<Functions> {
+    const groupings = this.state.initialized ? (
```

Comment:
`Groupings` should be its own component, especially since it has already been abstracted away from the rest of the presentational layer.